### PR TITLE
Added writing and reading for STM32WL option bytes

### PR DESCRIPTION
**Pull request by AlexKlimaj**
This PR adds reading and writing the option bytes on the STM32WL.

### CODE REVIEW

File: src/option_bytes.c [hunk 1]
```diff
@@ -472,6 +472,44 @@ static int stlink_write_option_bytes_h7(stlink_t *sl, uint8_t *base,
...
+                                        stm32_addr_t addr, uint32_t len) {
+  /* Write options bytes */
+  uint32_t val;
+  int ret = 0;
+  (void)len;
+  uint32_t data;
+
+  clear_flash_error(sl);
+
+  write_uint32((unsigned char *)&data, *(uint32_t *)(base));
```
Comment (Ant-ON) [inline src/option_bytes.c:492]: WL5x chips have size of option bytes area of a few words. I think it is better to add a loop write by word similar to H7 chips: https://github.com/stlink-org/stlink/blob/468b1d2daa853b975c33ab69876c486734f2c6a7/src/option_bytes.c#L424

**Review by Ant-ON — CHANGES_REQUESTED** [review]

File: src/option_bytes.c (hunk 1 above)
Comment (AlexKlimaj) [reply src/option_bytes.c]: Added the while loop

**Review by Ant-ON — APPROVED** [review]

**Comment by Ant-ON** [comment]
@AlexKlimaj May be before downgrade `RDP` need erase `WRP` bytes

**Comment by AlexKlimaj** [comment]
Here is the log from downgrading RDP in STM32CubeProgrammer. I don't see it erasing the WRP bytes.

```
11:18:12:235 : STLinkUSBDriver.dll loaded
11:18:12:249 : STLinkUSBDriver.dll loaded
11:18:12:249 : ST-LINK SN  : 50FF72067265575445271867
11:18:12:249 : ST-LINK FW  : V2J35S7
11:18:12:249 : Voltage     : 3.19V
11:18:12:250 : SWD freq    : 4000 KHz
11:18:12:250 : Connect mode: Normal
11:18:12:250 : Reset mode  : Software reset
11:18:12:333 : Device ID   : 0x497
11:18:12:503 : flash loader C:\Program Files\STMicroelectronics\STM32Cube\STM32CubeProgrammer\bin/FlashLoader/0x497.stldr is loaded
11:18:12:503 : Reading data...
11:18:12:506 : r ap 0 @0x1FFF75E0 0x00000004 bytes
11:18:12:610 : UPLOADING OPTION BYTES DATA ...
11:18:12:610 :   Bank          : 0x00
11:18:12:610 :   Address       : 0x58004020
11:18:12:610 :   Size          : 104 Bytes
11:18:12:611 : Reading data...
11:18:12:634 : r ap 0 @0x58004020 0x00000068 bytes
11:18:12:637 : UPLOADING OPTION BYTES DATA ...
11:18:12:637 :   Bank          : 0x00
11:18:12:637 :   Address       : 0x58004020
11:18:12:637 :   Size          : 104 Bytes
11:18:12:637 : Reading data...
11:18:12:637 : r ap 0 @0x58004020 0x00000068 bytes
11:18:12:652 : UPLOADING ...
11:18:12:652 :   Size          : 2456 Bytes
11:18:12:652 :   Address       : 0x8000000
11:18:12:652 : Read progress:
11:18:12:654 : Reading data...
11:18:12:654 : r ap 0 @0x08000000 0x00000998 bytes
11:18:12:654 : Error: Data read failed
11:18:17:268 : Option byte command : -ob  RDP=170 

11:18:17:281 : PROGRAMMING OPTION BYTES AREA ...
11:18:17:364 :   Bank          : 0x00
11:18:17:364 :   Address       : 0x58004020
11:18:17:364 :   Size          : 104 Bytes
11:18:17:364 : OB buffer: aaf0ff3fff000000000000007f0000007f000000ff00000000000000ffffffff00000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000000ff00ff800080fffe
11:18:17:364 : Buffer program...
11:18:17:448 : halt ap 0 
11:18:17:449 : w ap 0 reg 15 PC   (0x20000000)  
11:18:17:453 : w ap 0 reg 17 MSP  (0x20000500)  
11:18:17:453 : w ap 0 reg 16 xPSR (0x01000000)  
11:18:17:458 : w ap 0 @0x20002B80 0x00000200 bytes
11:18:17:459 : w ap 0 @0x20000000 0x00000004 bytes
11:18:17:541 : w ap 0 @0x20000004 0x0000274C bytes
11:18:17:544 : w ap 0 @0x20002B80 0x00000068 bytes
11:18:17:545 : Loader write option bytes...
11:18:17:545 : Init flashloader...
11:18:17:546 : halt ap 0 
11:18:17:549 : w ap 0 reg 0 R0   0x00000001
11:18:17:551 : w ap 0 reg 1 R1   0x00000000
11:18:17:556 : w ap 0 reg 2 R2   0x00000000
11:18:17:556 : w ap 0 reg 3 R3   0x00000000
11:18:17:556 : w ap 0 reg 4 R4   0x00000000
11:18:17:556 : w ap 0 reg 5 R5   0x00000000
11:18:17:556 : w ap 0 reg 6 R6   0x00000000
11:18:17:558 : w ap 0 reg 7 R7   0x00000000
11:18:17:560 : w ap 0 reg 8 R8   0x00000000
11:18:17:560 : w ap 0 reg 9 R9   0x00000000
11:18:17:560 : w ap 0 reg 10 R10  0x00000000
11:18:17:564 : w ap 0 reg 11 R11  0x00000000
11:18:17:564 : w ap 0 reg 12 R12  0x00000000
11:18:17:564 : w ap 0 reg 13 SP   0x00000000
11:18:17:564 : w ap 0 reg 14 LR   0x20000001
11:18:17:564 : w ap 0 reg 15 PC   0x2000002F
11:18:17:565 : w ap 0 reg 16 xPSR 0x01000000
11:18:17:565 : w ap 0 reg 17 MSP  0x20002B4C
11:18:17:565 : w ap 0 reg 18 PSP  0x00000000
11:18:17:565 : run ap 0 
11:18:17:579 : halt ap 0 
11:18:17:579 : r ap 0 reg 0 R0   0x00000001
11:18:17:582 : w ap 0 reg 0 R0   0x58004020
11:18:17:583 : w ap 0 reg 1 R1   0x00000068
11:18:17:583 : w ap 0 reg 2 R2   0x20002B80
11:18:17:586 : w ap 0 reg 3 R3   0x00000002
11:18:17:591 : w ap 0 reg 4 R4   0x00000000
11:18:17:592 : w ap 0 reg 5 R5   0x00000000
11:18:17:592 : w ap 0 reg 6 R6   0x00000000
11:18:17:603 : w ap 0 reg 7 R7   0x00000000
11:18:17:620 : w ap 0 reg 8 R8   0x00000000
11:18:17:620 : w ap 0 reg 9 R9   0x00000000
11:18:17:620 : w ap 0 reg 10 R10  0x00000000
11:18:17:620 : w ap 0 reg 11 R11  0x00000000
11:18:17:620 : w ap 0 reg 12 R12  0x00000000
11:18:17:620 : w ap 0 reg 13 SP   0x00000000
11:18:17:620 : w ap 0 reg 14 LR   0x20000001
11:18:17:621 : w ap 0 reg 15 PC   0x20000439
11:18:17:622 : w ap 0 reg 16 xPSR 0x01000000
11:18:17:623 : w ap 0 reg 17 MSP  0x20002B4C
11:18:17:623 : w ap 0 reg 18 PSP  0x00000000
11:18:17:623 : run ap 0 
11:18:17:693 : WriteOB function terminated with connection error due to OB_Launch
11:18:17:694 : Reconnecting...
11:18:18:766 : STLinkUSBDriver.dll loaded
11:18:18:766 : ST-LINK SN  : 50FF72067265575445271867
11:18:18:766 : ST-LINK FW  : V2J35S7
11:18:18:777 : Voltage     : 3.19V
11:18:18:777 : SWD freq    : 4000 KHz
11:18:18:777 : Connect mode: Normal
11:18:18:778 : Reset mode  : Software reset
11:18:18:858 : Device ID   : 0x497
11:18:18:858 : Reconnected !
11:18:18:858 : Success with Time elapsed during Reconnect: 94 ms
11:18:18:858 : UPLOADING OPTION BYTES DATA ...
11:18:18:858 :   Bank          : 0x00
11:18:18:858 :   Address       : 0x58004020
11:18:18:858 :   Size          : 104 Bytes
11:18:18:859 : Reading data...
11:18:18:891 : r ap 0 @0x58004020 0x00000068 bytes
11:18:18:891 : OPTION BYTE PROGRAMMING VERIFICATION:
11:18:18:891 : Option Bytes successfully programmed
11:18:19:064 : UPLOADING OPTION BYTES DATA ...
11:18:19:066 :   Bank          : 0x00
11:18:19:067 :   Address       : 0x58004020
11:18:19:067 :   Size          : 104 Bytes
11:18:19:068 : Reading data...
11:18:19:068 : r ap 0 @0x58004020 0x00000068 bytes

```

**Comment by AlexKlimaj** [comment]
I was able to write the option bytes by adding the ability to write to STM32WB_FLASH_OPTR.

**Review by Nightwalker-87 — APPROVED** [review]

**Pull request merged**